src/anm/analytics.js

```diff
@@ -0,0 +1,77 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics');
+
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
+            } else {
+                var trackUrl = animatronUrl + '?player=' + encodeURIComponent(data);
+                sendViaGif(trackUrl);
+            }
+        } else {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (trackUrl) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = trackUrl;
+        beacon.onerror = beacon.onload = function () {
+            beacon.onerror = beacon.onload = null;
+            setTimeout(event, timeout);
+        }
+    };
+    event();
```

Comment:
If I simplify your code, we'll get something like this:

```javascript
function Analytics() {
    var queue = [];
    function event() {
        if (!queue.empty()) {
            sendData();
            queue = [];
        } else {
            setTimeout(event, 1000);
        }
   }
   event();
   window.on('unload', event);
}
```

So, when you call `new Analytics()`, probably even before all scripts loaded into a page, you'll also manually call `event()` from inside. Since nothing yet happened to send, you just created the instance, and queue _for sure_ will be empty. So, it will set for 1sec or 2sec timeout to try to send data again (and, there's a high possibility that nothing meaningful will happen again for this second, page just loads).

So, the question is "Why?" :).